### PR TITLE
Refactor cli

### PR DESCRIPTION
**Pull request by jtenner**
There's a bunch of things here.

- [x] Readme clarifications
- [x] Make `run.ts` smaller by pulling out small functions
- [x] use `npm-run-all` to parallelize the typescript build
- [x] change `test.ts` to match `bin/asp`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
   "description": "Write AssemblyScript tests at blazing fast speeds",
   "main": "lib/index.js",
   "types": "./assembly/__tests__/as-pect.d.ts",
   "scripts": {
-    "test": "npm run tsc && node lib/test && jest",
-    "tsc": "tsc && tsc --outfile lib/as-pect.amd.js --module amd",
+    "test": "run-s tsc asp jest",
+    "asp": "node lib/test",
+    "jest": "jest",
+    "tsc": "run-p tsc:default tsc:amd",
```
Comment (willemneal) [inline package.json]: Nice!  Didn't know about that feature!

File: src/cli/util/collectPerformanceConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { IYargs } from "./IYargs";
+import { IPerformanceConfiguration } from "../../util/IPerformanceConfiguration";
+
+export function collectPerformanceConfiguration(yargs: IYargs, performanceConfiguration: IPerformanceConfiguration) {
+  if (yargs.argv.hasOwnProperty("performance"))
```
Comment (willemneal) [inline src/cli/util/collectPerformanceConfiguration.ts]: If this is false shouldn't you return from the function?  If you aren't going to collect performance data than why check every flag?

File: package.json (hunk 1 above)
Comment (jtenner) [reply package.json]: Yeah, it's a great feature. I use it for setting up development servers in complicated setups.

File: src/cli/util/collectPerformanceConfiguration.ts (hunk 2 above)
Comment (jtenner) [reply src/cli/util/collectPerformanceConfiguration.ts]: The `--performance` flag enables performance statistics gathering on *EVERY* test. This flag may not be present and it's still possible that `--collect-variance`, (for instance,) might be present. This should override what is currently in the `as-pect.config.js` file.

Comment (willemneal) [reply src/cli/util/collectPerformanceConfiguration.ts]: Ah okay.  I forgot about the programmatic way of recording stats.

**Pull request merged**